app/code/Magento/InventoryCatalog/Model/StockSourceLink/Validator/AssignToDefaultStockDefaultSourceValidator.php

```diff
@@ -51,9 +51,19 @@ public function __construct(
      */
     public function validate(StockSourceLinkInterface $link): ValidationResult
     {
+        $defaultStockId = $this->defaultStockProvider->getId();
+        $defaultSourceCode = $this->defaultSourceProvider->getCode();
+        $linkStockId = $link->getStockId();
+        $linkSourceCode = $link->getSourceCode();
+        $initialAssign = false;
+        if ($defaultStockId === $linkStockId
+        && $defaultSourceCode === $linkSourceCode) {
+            $initialAssign = true;
+        }
```

Comment:
Could you please move this logic into the dedicated private method 
`private function isAssignmentDefaultSourceOnDefaultStock`